src/main/kotlin/org/rust/lang/core/resolve/NameResolution.kt

```diff
@@ -219,11 +219,15 @@ fun processPathResolveVariants(lookup: ImplLookup, path: RsPath, isCompletion: B
     val crateRoot = path.crateRoot
     if (!path.hasColonColon) {
         if (Namespace.Types in ns) {
-            if (processor("self", containingMod)) return true
-            val superMod = containingMod.`super`
-            if (superMod != null) {
-                if (processor("super", superMod)) return true
+            val context = parent?.contextOfType<RsValueParameter>()
```

Comment:
Hm, I think `contextOfType` will overshoot here. What if we are completing something like

```
struct Foo;

fn foo(foo: se/*caret*/lf::Foo) {
}
```

Reply:
Hm, it seems to me that perhaps we should fix this at the level of insertion handler?

Reference completion for `self` parameter does not make much sense anyway, because it's more like a named element, and not a reference.

Reply:
I don't have time to look in depth until end of next week.

Reply:
I added both self and self:: to resolve this issue. I have no idea to fix it at the insertion handler.